js/src/behaviors/tabs.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * Tab control for assessment lens
+ * @param container - Container of tab control component
+ */
+/**
+ * Tabs behaviors
+ */
+export function initTabs() {
+	initTabNavClickListeners();
+	initTabControlClickListeners();
+
+	// update when tabs into the view
+	const tabContainers = Array.from(document.querySelectorAll('tab-container')) as HTMLElement[];
+	for (const tc of tabContainers) {
+		updateTabNav(tc);
+	}
+}
+
+function initTabNavClickListeners() {
+	window.addEventListener('click', (event: Event) => {
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('[data-tab-nav]') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('tab-container') as HTMLElement;
```

Comment:
data- attrs or web components for js functionality

Reply:
updated